omod/src/main/java/org/openmrs/module/coreapps/page/controller/MergeVisitsPageController.java

```diff
@@ -52,27 +64,54 @@ public Object controller(@RequestParam("patientId") Patient patient,
 
     public String post(@RequestParam("patientId") Patient patient,
                        @RequestParam("mergeVisits") List<Integer> mergeVisits,
-					   @RequestParam(value = "returnUrl", required = false) String returnUrl,
+                       @RequestParam(value = "returnUrl", required = false) String returnUrl,
                        @SpringBean AdtService service,
                        UiUtils ui,
-                       HttpServletRequest request, PageModel model){
+                       HttpServletRequest request, PageModel model) {
 
+        List<String> mergedVisitIds = new ArrayList<String>();
         if (patient.isVoided() || patient.isPersonVoided()) {
             return "redirect:" + ui.pageLink("coreapps", "patientdashboard/deletedPatient",
-					SimpleObject.create("patientId", patient.getId().toString() ));
+                    SimpleObject.create("patientId", patient.getId().toString()));
         }
 
-        if (mergeVisits!=null && mergeVisits.size() > 0 ){
+        if (mergeVisits != null && mergeVisits.size() > 0) {
             Visit mergedVisit = service.mergeConsecutiveVisits(mergeVisits, patient);
-            if (mergedVisit != null){
+            if (mergedVisit != null) {
                 request.getSession().setAttribute(AppUiConstants.SESSION_ATTRIBUTE_INFO_MESSAGE, ui.message("coreapps.task.mergeVisits.success"));
                 request.getSession().setAttribute(AppUiConstants.SESSION_ATTRIBUTE_TOAST_MESSAGE, "true");
-            }else{
+            } else {
                 request.getSession().setAttribute("emr.errorMessage", ui.message("coreapps.task.mergeVisits.error"));
                 request.getSession().setAttribute(AppUiConstants.SESSION_ATTRIBUTE_TOAST_MESSAGE, "true");
             }
+
+            //Pass by all visits except the first one.
+            for (int i = 1; i < mergeVisits.size(); i++) {
+                mergedVisitIds.add(mergeVisits.get(i).toString());
+            }
         }
+        return "redirect:" + ui.pageLink("coreapps", "mergeVisits", SimpleObject.create("patientId", patient.getId(), "returnUrl", returnUrl, "mergedVisitIds", mergedVisitIds));
+    }
+
 
-		return "redirect:" + ui.pageLink("coreapps", "mergeVisits", SimpleObject.create("patientId", patient.getId(), "returnUrl", returnUrl));
+    /**
+     * If some of the merged visit IDs are on the return url,  it scraps the visitId param altogether off the UR
+     * so it doesn't return to a visit that no longer exists.
+     */
+    public String updateVisitIdUrlParameter(String returnUrl, List<String> mergedVisitsIDs) {
```

Comment:
Can we have the method name reflect what kind of update it is doing?

Reply:
@dkayiwa what do you think about the name "removeVisitIdFromURLIfVisitWasMerged"